EIPS/eip-3525.md

```diff
@@ -0,0 +1,422 @@
+	- MUST revert if `_to` is the zero address.
+	- MUST revert if the transfer amount exceeds the actual amount of units in `_tokenId`.
+	- MUST revert if the transfer amount exceeds the VNFT approved units limit.
+	- MUST emit the VNFT level `PartialTransfer` event.
+
+**_safeTransferFrom rules:_**
+
+- `safeTransferFrom` SHOULD be used to implement the same function as `transferFrom`, with an extra step to check if the recipient is capable of receiving VNFTs by implementing the `onVNFTReceived` interface.
+- MUST obey the above rules set for `transferFrom`.
+- MUST check if `_to` is a smart contract (code size > 0). If so, `safeTransferFrom` MUST call `onVNFTReceived` on `_to` and MUST revert if the return value does not match `bytes4(keccak256("onVNFTReceived(address,address,uint256,uint256,bytes)"))`
+
+
+
+
+### Metadata 
+
+
+#### Metadata Extensions
+
+VNFT metadata extensions are compatible ERC721 metadata extensions.
+
+
+#### VNFT Metadata URI JSON Schema
+
+This is the "VNFT Metadata JSON Schema" referenced above.
+
+```json
+{
+    "title": "Asset Metadata",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "description": "Identifies the asset to which this NFT represents"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the asset to which this NFT represents"
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+        },
+         "decimals": {
+            "type": "integer",
+            "description": "The number of decimal places that the token amount should display - e.g. 18, means to divide the token amount by 1000000000000000000 to get its user representation."
+        },
+        "properties": {
+            "type": "object",
+            "description": "Arbitrary properties. Values may be strings, numbers, object or arrays."
+        }
+    }
+}
+```
+
+### Approval
+
+VNFT adds a new approval model, that is, one can approve operators to partial transfer units from a token with certain ID, the new interface is:
+```function approve(address to, uint256 tokenId, uint256 units); ```
+
+
+
+## Rationale 
+
+### Metadata generation
+
+Since VNFT is designed for representing underlying assets, rather than artifacts for gaming or arts, the implementation should give out the metadata directly from contract code, rather than give a URL of a server for returning metadata.
+
+
+### Design decision: Keep unsafe transfer
+
+There are mainly two reasons we keep the unsafe transfer interfaces:
+
+1. Since VNFT is ERC721 compatible, we must keep compatibility for all wallets and contracts that are still calling unsafe transfer interfaces for ERC721 tokens.
```

Comment:
```suggestion
1. Since VNFT is ERC-721 compatible, we must keep compatibility for all wallets and contracts that are still calling unsafe transfer interfaces for ERC-721 tokens.
```